server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -553,11 +552,23 @@ private void setTranslogFlushThresholdSize(ByteSizeValue byteSizeValue) {
     }
 
     private void setTranslogRetentionSize(ByteSizeValue byteSizeValue) {
-        this.translogRetentionSize = byteSizeValue;
+        if (softDeleteEnabled && byteSizeValue.getBytes() >= 0) {
+            logger.info("setting [{}] is ignored for translog is no longer used in peer recoveries with soft-deletes enabled",
```

Comment:
Should we use the deprecation logger here? We should also update the docs for these settings to detail when these settings are still in use.

Reply:
++. Will do